src/openvic-simulation/diplomacy/CountryRelation.hpp

```diff
@@ -46,50 +50,194 @@ namespace OpenVic {
+				second = std::bit_cast<uintptr_t>(pair.second);
+
+			std::size_t seed1(0);
+			OpenVic::utility::hash_combine(seed1, first);
+			OpenVic::utility::hash_combine(seed1, second);
 
 			std::size_t seed2(0);
-			OpenVic::utility::hash_combine(seed2, pair.second);
-			OpenVic::utility::hash_combine(seed2, pair.first);
+			OpenVic::utility::hash_combine(seed2, second);
+			OpenVic::utility::hash_combine(seed2, first);
 
 			return std::min(seed1, seed2);
 		}
 	};
 }
 
 namespace OpenVic {
-	using country_relation_value_t = int16_t;
-
 	struct CountryRelationManager {
-	private:
-		// TODO: reference of manager responsible for storing CountryInstances
-		ordered_map<CountryRelationPair, country_relation_value_t> country_relations;
+		using relation_value_type = int16_t;
+		class influence_value_type {
+			static constexpr uint16_t MAX_VALUE = 100; // TODO: implement defines.diplomacy.MAX_INFLUENCE
+			uint16_t value;
+
+		public:
+			constexpr influence_value_type() = default;
+			constexpr influence_value_type(uint16_t value) : value(std::min(value, MAX_VALUE)) {}
+
+#define BIN_OPERATOR(OP) \
+	template<typename T> \
+	friend inline constexpr influence_value_type operator OP(influence_value_type const& lhs, T const& rhs) { \
+		return std::min<uint16_t>(lhs.value OP rhs, MAX_VALUE); \
+	} \
+	template<typename T> \
+	friend inline constexpr influence_value_type operator OP(T const& lhs, influence_value_type const& rhs) { \
+		return std::min<uint16_t>(lhs OP rhs.value, MAX_VALUE); \
+	}
+
+			BIN_OPERATOR(+);
+			BIN_OPERATOR(-);
+			BIN_OPERATOR(*);
+			BIN_OPERATOR(/);
+			BIN_OPERATOR(%);
+
+#undef BIN_OPERATOR
+
+			template<typename T>
+			friend inline constexpr bool operator==(influence_value_type const& lhs, T const& rhs) {
+				return lhs.value == rhs;
+			}
+			template<typename T>
+			friend inline constexpr auto operator<=>(influence_value_type const& lhs, T const& rhs) {
+				return lhs.value <=> rhs;
+			}
 
-	public:
-		CountryRelationManager(/* TODO: Country Instance Manager Reference */);
 
-		bool add_country(CountryRelationInstanceProxy country);
-		bool remove_country(CountryRelationInstanceProxy country);
+#define BIN_OPERATOR_EQ(OP) \
+	template<typename T> \
+	friend inline constexpr influence_value_type& operator OP##=(influence_value_type & lhs, T const& rhs) { \
+		lhs.value = std::min<uint16_t>(lhs.value OP rhs, MAX_VALUE); \
+		return lhs; \
+	} \
+	friend inline constexpr influence_value_type& operator OP##=( \
+		influence_value_type & lhs, influence_value_type const& rhs \
+	) { \
+		return lhs OP## = rhs.value; \
+	}
+
+			BIN_OPERATOR_EQ(+);
+			BIN_OPERATOR_EQ(-);
+			BIN_OPERATOR_EQ(*);
+			BIN_OPERATOR_EQ(/);
+			BIN_OPERATOR_EQ(%);
+
+#undef BIN_OPERATOR_EQ
+		};
+
+		enum class OpinionType : int8_t {
+			Hostile = -2, //
+			Opposed,
+			Neutral,
+			Cordial,
+			Friendly,
+			Sphere
+		};
+
+	private:
+#define RELATION_PAIR_MAP(PAIR_TYPE, VALUE_TYPE, NAME, FUNC_NAME, recipient_CONST) \
```

Comment:
```suggestion
#define RELATION_PAIR_MAP(PAIR_TYPE, VALUE_TYPE, NAME, FUNC_NAME, RECIPIENT_CONST) \
```
Casing was ruined by replace all :/